packages/mocha-runner/src/mocha-runner.ts

```diff
@@ -135,7 +135,10 @@ class MochaRunner implements TestRunner {
             results.testResults = Util.filterTestResultsByTestLocator(results.testResults,
                 this._testlocator, this._blockListedLocators)
         }
-        mocha.dispose()
+        // removing testfile from cache to reload testfiles when new mocha instance is created
+        for (const testFile of testFilesToProcessList) {
```

Comment:
The function earlier had some additional check as well
https://sourcegraph.com/github.com/mochajs/mocha/-/blob/lib/mocha.js?L488
https://sourcegraph.com/github.com/mochajs/mocha/-/blob/lib/mocha.js?L479

we should use function version > 6, for rest we can use this one

Reply:
mocha dispose will call automatically after test execution. I called explicitly to bridge the time gap. Now using above method we are clearing the cache for test files. And later mocha framework dispose this object.

Reply:
Handled based on version using try/catch